### PR TITLE
Elasticsearch connector documentation fixes

### PR DESCRIPTION
**Pull request by fastcatch**
Fixing points 2, 4 and 5 in #12880.

### CODE REVIEW

**Comment by wenleix** [comment]
@zhenxiao : Can you help take a look at this PR? Thank you !

File: presto-docs/src/main/sphinx/connector/elasticsearch.rst [hunk 1]
```diff
@@ -263,11 +263,31 @@ Optionally, column metadata can be described in the same table description JSON
...
+``binary``    ``VARBINARY``
+``boolean``   ``BOOLEAN``
+``double``    ``DOUBLE``
+``float``     ``DOUBLE``
+``integer``   ``INTEGER``
+``keyword``   ``VARCHAR``
+``long``      ``BIGINT``
+``string``    ``VARCHAR``
+``text``      ``VARCHAR``
+(all others)  (unsupported)
```
Comment (zhenxiao) [inline presto-docs/src/main/sphinx/connector/elasticsearch.rst]: s/all others/others/g

File: presto-docs/src/main/sphinx/connector/elasticsearch.rst [hunk 2]
```diff
@@ -263,11 +263,31 @@ Optionally, column metadata can be described in the same table description JSON
...
-``jsonPath``          optional  string         Json path of Elasticsearch field.
-``jsonType``          optional  string         Json type of Elasticsearch field.
+``name``              required  string         Column name of Elasticsearch field.
+``type``              required  string         Column type of Elasticsearch field (see second column of `data type mapping`_).
+``jsonPath``          required  string         Json path of Elasticsearch field (when in doubt set to the same as ``name``).
+``jsonType``          required  string         Json type of Elasticsearch field (when in doubt set to the same as ``type``).
 ``ordinalPosition``   optional  integer        Ordinal position of the column.
 ===================== ========= ============== =============================
 
-.. _field: https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-types.html
```
Comment (zhenxiao) [inline presto-docs/src/main/sphinx/connector/elasticsearch.rst]: prefer keep this link, as users might not be familiar with elasticsearch type

File: presto-docs/src/main/sphinx/connector/elasticsearch.rst [hunk 3]
```diff
@@ -263,11 +263,31 @@ Optionally, column metadata can be described in the same table description JSON
 ===================== ========= ============== =============================
 Field                 Required  Type           Description
 ===================== ========= ============== =============================
-``name``              optional  string         Column name of Elasticsearch field.
-``type``              optional  string         Column type of Elasticsearch `field`_.
-``jsonPath``          optional  string         Json path of Elasticsearch field.
-``jsonType``          optional  string         Json type of Elasticsearch field.
+``name``              required  string         Column name of Elasticsearch field.
+``type``              required  string         Column type of Elasticsearch field (see second column of `data type mapping`_).
```
Comment (zhenxiao) [inline presto-docs/src/main/sphinx/connector/elasticsearch.rst]: prefer keep the field link, and point to type mapping

File: presto-docs/src/main/sphinx/connector/elasticsearch.rst [hunk 4]
```diff
@@ -263,11 +263,31 @@ Optionally, column metadata can be described in the same table description JSON
...
+``jsonPath``          required  string         Json path of Elasticsearch field (when in doubt set to the same as ``name``).
```
Comment (zhenxiao) [inline presto-docs/src/main/sphinx/connector/elasticsearch.rst]: could we use words other than 'when in doubt'? maybe default as name?

File: presto-docs/src/main/sphinx/connector/elasticsearch.rst [hunk 5]
```diff
@@ -263,11 +263,31 @@ Optionally, column metadata can be described in the same table description JSON
...
+``jsonType``          required  string         Json type of Elasticsearch field (when in doubt set to the same as ``type``).
```
Comment (zhenxiao) [inline presto-docs/src/main/sphinx/connector/elasticsearch.rst]: same as jsonType

**Pull request closed without merging**